library/core/src/intrinsics.rs

```diff
@@ -2793,6 +2752,150 @@ pub unsafe fn vtable_align(_ptr: *const ()) -> usize {
     unreachable!()
 }
 
+/// The size of a type in bytes.
+///
+/// Note that, unlike most intrinsics, this is safe to call;
+/// it does not require an `unsafe` block.
+/// Therefore, implementations must not require the user to uphold
+/// any safety invariants.
+///
+/// More specifically, this is the offset in bytes between successive
+/// items of the same type, including alignment padding.
+///
+/// The stabilized version of this intrinsic is [`core::mem::size_of`].
+#[rustc_nounwind]
+#[unstable(feature = "core_intrinsics", issue = "none")]
+#[rustc_const_stable(feature = "const_size_of", since = "1.40.0")]
+#[rustc_intrinsic]
+#[rustc_intrinsic_must_be_overridden]
+#[cfg(not(bootstrap))]
+pub const fn size_of<T>() -> usize {
+    unreachable!()
+}
```

Comment:
If I understand correctly, it looks like `rustc_intrinsic_must_be_overridden` isn't needed if there is a fallback body https://doc.rust-lang.org/stable/unstable-book/language-features/intrinsics.html#rustc_intrinsic-declarations. Not sure what the difference is.

Reply:
The fallback body is never meant to be used, so the attribute is right. Only intrinsics with behaviour preserving bodies should not have that attribute